samples/template/src/main.c

```diff
@@ -0,0 +1,517 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <sid_api.h>
+#include <sid_hal_reset_ifc.h>
+#include <pal_init.h>
+#include <app_ble_config.h>
+#include <app_subGHz_config.h>
+#include <dk_buttons_and_leds.h>
+#include <zephyr/kernel.h>
+#include <zephyr/smf.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(app, CONFIG_LOG_DEFAULT_LEVEL);
+
+static struct k_thread app_thread;
+K_THREAD_STACK_DEFINE(app_thread_stack, CONFIG_SIDEWALK_THREAD_STACK_SIZE);
+
+typedef enum app_events {
+	APP_EVENT_SIDEWALK,
+	APP_EVENT_ERROR,
+	APP_EVENT_READY,
+	APP_EVENT_NOT_READY,
+	APP_EVENT_SEND,
+	APP_EVENT_FACTORY_RESET,
+	APP_EVENT_LINK_SWITCH
+} app_event_t;
+
+enum app_state {
+	STATE_RUNNING,
+	STATE_INIT,
+	STATE_SIDEWALK_NOT_READY,
+	STATE_SIDEWALK_CONNECTING,
+	STATE_SIDEWALK_READY,
+};
+
+struct app_sm {
+	struct smf_ctx ctx;
+	struct k_msgq msgq;
+	app_event_t event;
+};
+
+struct sid_ctx_s {
+	struct app_sm sm;
+	struct sid_handle *handle;
+	struct sid_config config;
+};
+
+struct app_msg {
+	struct sid_msg msg;
+	struct sid_msg_desc desc;
+};
+
+static int app_event_send(app_event_t event);
+
+static void state_running_run(void *o);
+static void state_init_entry(void *o);
+static void state_init_run(void *o);
+static void state_ready_run(void *o);
+static void state_not_ready_run(void *o);
+static void state_connecting_entry(void *o);
+static void state_connecting_run(void *o);
+
+static const struct smf_state app_states[] = {
+	[STATE_RUNNING] = SMF_CREATE_STATE(NULL, state_running_run, NULL, NULL),
+	[STATE_INIT] = SMF_CREATE_STATE(state_init_entry, state_init_run, NULL,
+					&app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_READY] =
+		SMF_CREATE_STATE(NULL, state_ready_run, NULL, &app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_NOT_READY] =
+		SMF_CREATE_STATE(NULL, state_not_ready_run, NULL, &app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_CONNECTING] = SMF_CREATE_STATE(state_connecting_entry, state_connecting_run,
+						       NULL, &app_states[STATE_RUNNING]),
+};
+
+static uint8_t echo_payload[255];
```

Comment:
For initial revision, OK, but I would ideally replace this by a message queue, with variable size of queue.
This is an optimization for a feature that has not yet been needed, so maybe we should stick with single message buffer, as a simple solution for now.

Reply:
nice to have